lib/serial_terminal.pm

```diff
@@ -96,6 +96,10 @@ sub login {
         type_string("\n");
         die 'Failed to confirm that login was successful' unless wait_serial(qr/$escseq* \w+:~\s\# $escseq* \s*$/x);
     }
+    else {
+        # wait for shell to avoid PS1 failures on passwordless login e.g. LIVECD
+        die 'Failed to confirm that passwordless login was successful' unless wait_serial(qr/.*:~\s\#/);
```

Comment:
AFAICT this also matches for the case above, so just move that out?

Reply:
This is fixing this failure when no password is needed, without waiting for shell it will immediately after login type to set PS1 which will fail as the shell is not ready.

https://openqa.opensuse.org/tests/1263528#step/consoletest_setup/3
```
[2020-05-12T05:57:33.206 CEST] [debug] >>> testapi::wait_serial: (?^ui:login:\s*$): ok
[2020-05-12T05:57:33.206 CEST] [debug] tests/console/consoletest_setup.pm:34 called opensusebasetest::select_serial_terminal -> lib/opensusebasetest.pm:1113 called testapi::select_console -> lib/susedistribution.pm:746 called serial_terminal::login -> lib/serial_terminal.pm:92 called testapi::type_string
[2020-05-12T05:57:33.206 CEST] [debug] <<< testapi::type_string(text="root\n")
[2020-05-12T05:57:33.206 CEST] [debug] <<< consoles::serial_screen::type_string(json_cmd_token="KfkNBVNu", text="root\n", cmd="backend_type_string")
[2020-05-12T05:57:33.207 CEST] [debug] tests/console/consoletest_setup.pm:34 called opensusebasetest::select_serial_terminal -> lib/opensusebasetest.pm:1113 called testapi::select_console -> lib/susedistribution.pm:746 called serial_terminal::login -> lib/serial_terminal.pm:99 called testapi::type_string
[2020-05-12T05:57:33.207 CEST] [debug] <<< testapi::type_string(text="PS1=\"# \"\n")
[2020-05-12T05:57:33.207 CEST] [debug] <<< consoles::serial_screen::type_string(json_cmd_token="hjQGqxgt", text="PS1=\"# \"\n", cmd="backend_type_string")
[2020-05-12T05:57:33.207 CEST] [debug] tests/console/consoletest_setup.pm:34 called opensusebasetest::select_serial_terminal -> lib/opensusebasetest.pm:1113 called testapi::select_console -> lib/susedistribution.pm:746 called serial_terminal::login -> lib/serial_terminal.pm:100 called testapi::wait_serial
```

Reply:
I know, but this would just work just fine AFAICT:

```
if (length $testapi::password) {
    ....
    type_string("\n");
}

die 'Failed to confirm that login was successful' unless wait_serial(qr/.*:~\s\#/);
```

Reply:
yes, I will do that.

Reply:
done